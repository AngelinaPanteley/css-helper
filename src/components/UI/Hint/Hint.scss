@import '../../../variables';

.Hint {
  padding: 10px 50px;
  position: fixed;
  top: 79px;
  left: 0px;
  transition: $transition;
  color: $color-white;
  min-height: 36px;

  &.Open {
    transform: translateX(0);
  }

  &.Close {
    transform: translateX(-100%);
  }
}

.Cross {
  position: absolute;
  top: -2px;
  right: -9px;
  transform: rotate(45deg);
  color: $color-white;
  font-size: 15px;
  background: none;
  border: none;
  outline: none;
  cursor: pointer;
}