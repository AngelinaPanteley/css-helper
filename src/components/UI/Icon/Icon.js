import React from 'react';

import styles from './Icon.scss';

const icon = (props) => {
  return (
    <i className={`${styles.icon} ${styles['icon_' + props.icon]}`} />
  )
}

export default icon;