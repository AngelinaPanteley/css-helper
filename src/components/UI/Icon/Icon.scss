@font-face {
  font-family: 'css-helper';
  src: url('../../../assets/icons/font/css-helper.eot?61525200');
  src: url('../../../assets/icons/font/css-helper.eot?61525200#iefix') format('embedded-opentype'),
       url('../../../assets/icons/font/css-helper.woff2?61525200') format('woff2'),
       url('../../../assets/icons/font/css-helper.woff?61525200') format('woff'),
       url('../../../assets/icons/font/css-helper.ttf?61525200') format('truetype'),
       url('../../../assets/icons/font/css-helper.svg?61525200#css-helper') format('svg');
  font-weight: normal;
  font-style: normal;
}
/* Chrome hack: SVG is rendered more smooth in Windozze. 100% magic, uncomment if you need it. */
/* Note, that will break hinting! In other OS-es font will be not as sharp as it could be */
/*
@media screen and (-webkit-min-device-pixel-ratio:0) {
  @font-face {
    font-family: 'css-helper';
    src: url('../font/css-helper.svg?61525200#css-helper') format('svg');
  }
}
*/
 
.icon:before {
  font-family: "css-helper";
  font-style: normal;
  font-weight: normal;
  speak: none;
 
  display: inline-block;
  text-decoration: inherit;
  width: 1em;
  margin-right: .2em;
  text-align: center;
  /* opacity: .8; */
 
  /* For safety - reset parent styles, that can break glyph codes*/
  font-variant: normal;
  text-transform: none;
 
  /* fix buttons height, for twitter bootstrap */
  line-height: 1em;
 
  /* Animation center compensation - margins should be symmetric */
  /* remove if not needed */
  margin-left: .2em;
 
  /* you can be more comfortable with increased icons size */
  /* font-size: 120%; */
 
  /* Font smoothing. That was taken from TWBS */
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
 
  /* Uncomment for 3D effect */
  /* text-shadow: 1px 1px 1px rgba(127, 127, 127, 0.3); */
}
 
.icon_down:before { content: '\e800'; } /* '' */
.icon_left:before { content: '\e801'; } /* '' */
.icon_right:before { content: '\e802'; } /* '' */
.icon_up:before { content: '\e803'; } /* '' */
.icon_edit:before { content: '\e804'; } /* '' */
.icon_delete:before { content: '\e805'; } /* '' */
.icon_plus:before { content: '\e806'; } /* '' */
.icon_minus:before { content: '\e807'; } /* '' */
.icon_download:before { content: '\e808'; } /* '' */
.icon_upload:before { content: '\e809'; } /* '' */
.icon_menu:before { content: '\f008'; } /* '' */
.icon_copy:before { content: '\f0c5'; } /* '' */
.icon_account:before { content: '\f2be'; } /* '' */