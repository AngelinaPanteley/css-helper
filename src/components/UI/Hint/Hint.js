import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import styles from './Hint.scss';
import Icon from '../Icon/Icon';

class Hint extends PureComponent {
  static propTypes = {
    isOpen: PropTypes.bool.isRequired,
    isError: PropTypes.bool,
    onClose: PropTypes.func.isRequired,
  }

  render() {
    const hintStyles = [styles.Hint];
    if (this.props.isOpen) {
      hintStyles.push(styles.Open);
      this.timer = setTimeout(this.props.onClose, 2000);
    } else {
      hintStyles.push(styles.Close);
      clearTimeout(this.timer);
    }
    return (
      <div className={hintStyles.join(' ')}
        style={{ background: this.props.isError ? '#bf1039' : '#19b319' }}>
        {this.props.entryText}
        <button className={styles.Cross} onClick={this.props.onClose}>
          <Icon icon='plus' />
        </button>
      </div>
    )
  }
}

export default Hint;