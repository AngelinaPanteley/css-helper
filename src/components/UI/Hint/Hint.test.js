"use strict";

import React from 'react';
import renderer from 'react-test-renderer';
import { expect as chaiExpect } from 'chai';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16.1';
import styles from './Hint.scss';
import Hint from './Hint.js';
import Icon from '../../UI/Icon/Icon.js'

configure({ adapter: new Adapter() });

describe('>>>Hint', () => {

  //SNAPSHOTS

  const props = {
    isOpen: true,
    isError: true,
    onClose: () => { },
  };

  const hintComponent = <Hint {...props} />;

  const component = renderer.create(hintComponent);

  it('renders save hint correctly', () => {
    let componentTree = component.toJSON();
    expect(componentTree).toMatchSnapshot();
  });

  //SHALLOW RENDERING

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(hintComponent);
  });

  it('renders close button with cross icon', () => {
    chaiExpect(wrapper.contains(
      <button className={styles.Cross} onClick={props.onClose}>
        <Icon icon='plus' />
      </button>
    )).to.equal(true);
  });
})