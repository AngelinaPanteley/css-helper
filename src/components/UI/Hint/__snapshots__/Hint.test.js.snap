// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>Hint renders save hint correctly 1`] = `
<div
  className=" "
  style={
    Object {
      "background": "#bf1039",
    }
  }
>
  <button
    onClick={[Function]}
  >
    <i
      className="undefined undefined"
    />
  </button>
</div>
`;
